torch/csrc/autograd/engine.cpp

```diff
@@ -79,10 +84,12 @@ struct CompareFunctionTaskTime {
       return false;
     } else if (!t2.fn_) {
       return true;
-    } else {
+    } else if (t1.base_== t2.base_) {
       return t1.fn_->sequence_nr() < t2.fn_->sequence_nr();
+    } else {
+      return t1.getReentrantDepth() <= t2.getReentrantDepth();
```

Comment:
Hmm. But if the reentrant depths match, don't I still want to test on sequence numbers? So it seems like if there is a difference in reentrant depth, you should use that, otherwise you should defer to sequence number comparison here.